File: src/services/instructions/loadstore/LD_S_REGX_MREGY.java
package emulator.source.loadstore

import emulator.engine.CpuContext
 import Instruction from '../Instruction'import REGISTER_VALUE_NAME_MAPPER from '@/constants/registers'

export default class LD_S_REGX_MREGY extends Instruction {
	public LD_S_REGX_MREGY(memory, address, source, 
 destination) {
		super(memory, address, source, destination)
		super.setAssembler("ld.s " + this.sdest + ", [" + this.ssource + "]")
	}

	
	exec ({ context, memory }) {
		context.getReg(this.destination)  = context.memory[context.getReg(this.source)  / 2] & 0xFFFF
		context.pc  += 2
	}
}

File: src/services/instructions/loadstore/LdSRegXMRegY.js
import Instruction from '../Instruction'
import { REGISTER_VALUE_NAME_MAPPER } from '@/constants/registers'

const ASSEMBLER_INSTRUCTION_EXPRESSION = (sdestination, ssource) =>
  `ld.s ${sdestination}, [${ssource}]`

export default class LdSRegXMRegY extends Instruction {
  constructor(memory, address, source, destination, symbolTable) {
    super(memory, address, source, destination, symbolTable)
    super.setAssembler(
      ASSEMBLER_INSTRUCTION_EXPRESSION(this.sdestination, this.ssource)
    )
  }

  exec({ context, memory }) {
    context[REGISTER_VALUE_NAME_MAPPER[this.destination]] =
      memory[context[REGISTER_VALUE_NAME_MAPPER[this.source]] / 2] & 0xffff
    context.pc += 2
  }
}
